extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -226,8 +246,58 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
     const attributes = /** @type {!JsonObject} */ (Object.assign({},
         defaultAttrs, configAttrs));
 
-    return createElementWithAttributes(
+    const adElement = createElementWithAttributes(
         document, 'amp-ad', attributes);
+
+    this.currentAdElement_ = adElement;
+    return adElement;
+  }
+
+
+  /**
+   * Validate ad-server response has requirements to build outlink
+   * @param {!Element} adPageElement
+   */
+  maybeCreateCtaLayer_(adPageElement) {
+    // if making a CTA layer we need a button name & outlink url
+    const ctaUrl = this.currentAdElement_.getAttribute(CTA_URL_ATTR);
+    const ctaType = this.currentAdElement_.getAttribute(CTA_TYPE_ATTR);
+
+    if (!ctaUrl || !ctaType) {
+      dev().error(TAG, 'Both "vars.ctaType" & "vars.ctaUrl" ' +
+          'are required in ad-server JSON response."');
+    }
+
+    const ctaText = CTA_TYPES[ctaType];
+    dev().assert(ctaText, `${TAG}: the 'ctaType' returned by the ad-server` +
+        'must be one of the predefined choices.');
+    return this.createCtaLayer_(adPageElement, ctaText, ctaUrl);
+  }
+
+
+  /**
+   * Create layer to contain outlink button
+   * @param {!Element} adPageElement
+   * @param {string} ctaText
+   * @param {string} ctaUrl
+   */
+  createCtaLayer_(adPageElement, ctaText, ctaUrl) {
+    const button = document.createElement('button');
+    button.className = 'i-amphtml-story-ad-button';
+
+    const span = document.createElement('span');
+    span.className = 'i-amphtml-story-ad-button-text';
+    span.textContent = ctaText;
+
+    const gestures = Gestures.get(button, /* shouldNotPreventDefault */ false);
+    gestures.onGesture(TapRecognizer, () => {
```

Comment:
as discussed offline, using `<a>` will have a couple of advantages.

Reply:
done